zerver/webhooks/sentry/view.py

```diff
@@ -81,6 +81,20 @@
 }
 
 
+def is_sample_event(event: Dict[str, Any]) -> bool:
+    # These are just some workarounds; there is no documented way to check
+    # if an event is a sample event or not.
+    tags = event.get("tags", [])
+    if ["sample_event", "yes"] in tags:
+        return True
+    title = event.get("title", "")
+    if title == "This is an example Python exception":
+        return True
```

Comment:
I'd love a mention in here of why we have both of these checks, since they seem potentially redundant unless one knows the history here.

Reply:
You are right the checks are redundant. I would remove the `tags` check and just keep the `title` check for sample events.

> Does the Sentry UI show the response body from the server when you push the "test" button? 

When the integration is set through Alerts there is no response body shown to the user but when it is set through the Webhook plugin a response body is shown only when an error is returned. So I don't think we can hint the user as such. I think since it's clearly mentioned in the [documentation](https://zulip.com/integrations/doc/sentry) that setting the integration through the Webhook plugin is not the best idea hence, there is no need for this.

>  when they're setting it up via alerts, which may not provide the full datastream

I think you meant to say via the webhook plugin because if someone sets an integration through that that's when the integration doesn't provide all the events.

Reply:
I actually wasn't suggesting that you remove the other check -- I said they "seem potentially redundant unless one knows the history here."  I feel that, if we could only use it, the tags solution is better because it's less likely to have false-positives.  I can see users typing `raise("This is an example Python exception")` in their code when trying it out, and I'd rather that we not _think_ it's s sample-from-Sentry.  Right now, the only effect is that we'd accept a Raven SDK submission, but code does drift and if we, say, started doing other things based on `is_sample_event`, false-positives would get more dangerous.

Since we _have_ to use the title, I'm fine with dropping the tag support, but I'd prefer to adjust the "This is just a workaround" comment to make clearer that this is a _heuristic_ which may false-positive, so we should not rely on it for important behaviour decisions.

> When the integration is set through Alerts there is no response body shown to the user but when it is set through the Webhook plugin a response body is shown only when an error is returned. So I don't think we can hint the user as such.

Bummer.

> I think since it's clearly mentioned in the [documentation](https://zulip.com/integrations/doc/sentry) that setting the integration through the Webhook plugin is not the best idea hence, there is no need for this.

It's documented, but _people will not always read the documentation_ -- which is what made us need this PR in the first place. :)  So we should always think about if it's simple to have the user interface (here, through the Sentry UI) give good feedback.  So while there's no "need" for this, in that it's not a blocker, it certainly would have improved the product experience if we could.

> I think you meant to say via the Webhook plugin because if someone sets an integration through that that's when the integration doesn't provide all the events.

Ah, yeah, I said that backwards.  Sorry for any confusion!

Reply:
> Since we have to use the title, I'm fine with dropping the tag support, but I'd prefer to adjust the "This is just a workaround" comment to make clearer that this is a heuristic which may false-positive, so we should not rely on it for important behaviour decisions.

Yes, I think that makes perfect sense.

>So while there's no "need" for this, in that it's not a blocker, it certainly would have improved the product experience if we could.

You are correct that would have certainly improved the overall experience. Since there is no way right now to do that, I feel we can leave this for now.